fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/event/creation/RollCallEventCreationFragment.java

```diff
@@ -0,0 +1,151 @@
+    private LaoDetailViewModel mLaoDetailViewModel;
+    private EditText rollCallTitleEditText;
+    private Button confirmButton;
+    private Button openButton;
+
+  private final TextWatcher confirmTextWatcher =
+      new TextWatcher() {
+        @Override
+        public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+        @Override
+        public void onTextChanged(CharSequence s, int start, int before, int count) {
+          String meetingTitle = rollCallTitleEditText.getText().toString().trim();
+          boolean areFieldsFilled =
+              !meetingTitle.isEmpty() && !getStartDate().isEmpty() && !getStartTime().isEmpty();
+          confirmButton.setEnabled(areFieldsFilled);
+          openButton.setEnabled(areFieldsFilled);
+        }
+
+        @Override
+        public void afterTextChanged(Editable s) {}
+      };
+
+  public static RollCallEventCreationFragment newInstance() {
+    return new RollCallEventCreationFragment();
+  }
+
+  @Nullable
+  @Override
+  public View onCreateView(
+      @NonNull LayoutInflater inflater,
+      @Nullable ViewGroup container,
+      @Nullable Bundle savedInstanceState) {
+
+      mFragBinding = FragmentCreateRollCallEventBinding.inflate(inflater, container, false);
+
+      mLaoDetailViewModel = LaoDetailActivity.obtainViewModel(getActivity());
+
+      setDateAndTimeView(mFragBinding.getRoot(), this, getFragmentManager());
+      addDateAndTimeListener(confirmTextWatcher);
+
+      rollCallTitleEditText = mFragBinding.rollCallTitleText;
+      openButton = mFragBinding.rollCallOpen;
+      confirmButton = mFragBinding.rollCallConfirm;
+
+      mFragBinding.setLifecycleOwner(getActivity());
+
+      return mFragBinding.getRoot();
+  }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        setupConfirmButton();
+        setupOpenButton();
+        setupCancelButton();
+
+        // Subscribe to "new LAO event creation" event
+        mLaoDetailViewModel
+                .getNewLaoEventCreationEvent()
+                .observe(
+                        this,
+                        eventTypeEvent -> {
+                            EventType eventType = eventTypeEvent.getContentIfNotHandled();
+                            if(eventType == EventType.ROLL_CALL) {
+                                createRollCall();
+                            }
+                        }
+                );
+
+        // Subscribe to "open new roll call" event
+        mLaoDetailViewModel
+                .getOpenNewRollCallEvent()
+                .observe(
+                        this,
+                        booleanEvent -> {
+                            Boolean action = booleanEvent.getContentIfNotHandled();
+                            if(action != null) {
+                                String id = createRollCall();
+                                openRollCall(id);
```

Comment:
isn't it a bit redundant to call createRollCall() a second time when it was already called in the subscribe to "new LAO event creation " ?
I don't know if something  like 
final String[] RollCallId = new String[1];
        // Subscribe to "new LAO event creation" event
        mLaoDetailViewModel
                .getNewLaoEventCreationEvent()
                .observe(
                        this,
                        eventTypeEvent -> {
                            EventType eventType = eventTypeEvent.getContentIfNotHandled();
                            if(eventType == EventType.ROLL_CALL) {
                               RollCallId[0] =  createRollCall();
                            }
                        }
                );

        // Subscribe to "open new roll call" event
        mLaoDetailViewModel
                .getOpenNewRollCallEvent()
                .observe(
                        this,
                        booleanEvent -> {
                            Boolean action = booleanEvent.getContentIfNotHandled();
                            if(action != null) {
                                openRollCall(RollCallId[0]);
                            }
                        }
                );
would be any better ?

Reply:
The method `newLaoEventCreation` is only called when the confirm button is clicked. When the open button is clicked we will only receive a `openNewRollCallEvent`, and not a `newLaoEventCreationEvent`, because the only method called is `openNewRollCall`. So we will only call `openRollCall` once when we want to open a new roll call

Reply:
Good question, I can see that renaming the variables and adding some comments might help understand the difference between `mNewLaoEventEvent` and `mNewLaoEventCreationEvent`.

- `mNewLaoEventEvent ` represents the action of choosing an event type to create on the multiple-choice screen. 
- `mNewLaoEventCreationEvent` represent the action of creating an event on the event creation screen, for example when we click the button confirm on the roll call creation screen.

As a solution I propose to rename `mNewLaoEventEvent` to `mChooseNewLaoEventTypeEvent` and `addEvent` to `chooseEventType`. What do you think?

Reply:
Yes this is a good idea I think these new names would make more sense.